sdk/client-core/core/src/main/java/io/clientcore/core/http/models/ServerSentEventListener.java

```diff
@@ -44,7 +44,7 @@ default void onError(Throwable throwable) {
      * @param lastEventId ID of last event that was received
      * @return whether to retry
      */
-    default boolean shouldRetry(Throwable throwable, Duration retryAfter, long lastEventId) {
+    default boolean shouldRetry(Throwable throwable, Duration retryAfter, String lastEventId) {
```

Comment:
It feels a little odd that a listener is being made responsible for determining if a retry should happen?

Reply:
I was thinking if users based on the
- exception type or
-  knowing if they have a last event id present or 
-  retry after not being too long, 
wanted to give it a retry.

We can always default the above behavior and  "honor" the "retry/reconnect time", remove this API altogether?

Reply:
yeah, I wonder if we take it out for now and do default behavior. We can always add it back later in a better shape, if needed.

Reply:
made it default.